lib/ash/resource/transformers/belongs_to_attribute.ex

```diff
@@ -23,7 +23,12 @@ defmodule Ash.Resource.Transformers.BelongsToAttribute do
       case Transformer.build_entity(@extension, [:attributes], :attribute,
              name: relationship.source_field,
              type: relationship.field_type,
-             allow_nil?: not relationship.required?,
+             allow_nil?:
+               if relationship.primary_key? do
```

Comment:
Here we should return an error, e.g `{:error, "Belongs to relationship cannot be a primary key unless it is also marked as required}`. By setting it to `false` here, we'd override their resource, and they could have `belongs_to :foo, Bar, required?: false` not realizing that it is being ignored.

Reply:
Ok